### DISCUSSION TITLE
Add `sb update` command for health-checking existing storybooks

### DISCUSSION THREAD

**Discussion by kroeder** [storybookjs/storybook#23033] [Ideas]
## Idea
Add a command `sb update`
- --version={valid npm/yarn version tag} // default: `sb update --version=latest`
- --skip-install // skipping yarn/npm install

## What it does
- Bump storybook versions (check package.json for @storybook/... and update version)
- Checks, if everything `sb init` would do, is correctly applied
  - For e.g. angular projects: 
    - Is there a custom `.storybook/tsconfig.json` which is required per default
    - Are `**/*.stories.ts` excluded in the `defaultProject` (usually `src/tsconfig.app.json`)
- It's basically `sb init` but works kinda as a health-check for the current installation

## Why?
Everything we update in `sb init` and the cli-templates won't get to developers who already have an existing project with storybook. If they run in already fixed bugs they have to either search for the fix in our issues or are going to create new issues. Updating storybook should be as easy as `sb init`.

## Who to contact
@Keraito: @igor-dv told me you might wanna join this discussion?

**Comment by kroeder** [comment]
Might has to be something interactive with user input because `sb update` should not overwrite things the user does not want. In cases he did lots of custom stuff

```
> Do you want to update .storybook/tsconfig.json? (y/n)
```

Or at least add `sb update --only-dependencies` as a tool to bump all versions but nothing else

**Comment by kroeder** [comment]
I also had the idea to add `warnings` on `start-storybook` for missing configurations in the latest updates.
Maybe this can be integrated in `start-storybook` with an info to run `sb update` in order to fix the misconfiguration

**Comment by shilman** [comment]
Converting this to a discussion. We’re reserving feature issues for things that are on our [short-term roadmap](https://github.com/storybookjs/storybook/milestones). We’ll check back periodically to see whether this is a good candidate to be added. If this feature would be useful to you, please upvote!
